examples/erc20/tests/erc20.rs

```diff
@@ -1216,3 +1256,237 @@ async fn error_when_transfer_from(alice: Account, bob: Account) -> Result<()> {
 
     Ok(())
 }
+
+// ============================================================================
+// Integration Tests: ERC-20 Permit Extension
+// ============================================================================
+
+#[e2e::test]
+async fn error_when_expired_deadline_for_permit(
+    alice: Account,
+    bob: Account,
+) -> Result<()> {
+    let contract_addr = deploy(alice.url(), &alice.pk(), None).await?;
+    let contract_alice = Erc20::new(contract_addr, &alice.wallet);
+    let alice_addr = alice.address();
+    let bob_addr = bob.address();
+
+    let balance = uint!(10_U256);
+    let _ = watch!(contract_alice.mint(alice_addr, balance))?;
+
+    let struct_hash = prepare_struct_hash(
+        alice_addr,
+        bob_addr,
+        balance,
+        U256::ZERO,
+        EXPIRED_DEADLINE,
+    );
+
+    let typed_data_hash =
+        prepare_typed_data_hash(domain_separator!(contract_alice), struct_hash);
+    let signature = alice
+        .sign_hash(&alloy::primitives::B256::from_slice(
+            typed_data_hash.as_slice(),
+        ))
+        .await;
+
+    let err = send!(contract_alice.permit(
+        alice_addr,
+        bob_addr,
+        balance,
+        EXPIRED_DEADLINE,
+        signature.v().y_parity_byte_non_eip155().unwrap(),
```

Comment:
Are you sure this is correct? I think this should be `signature.v().to_u64() as u8` instead, to account for all transaction types.

Reply:
Yep, we require `v` to be either `27` or `28` following behaviour from Solidity -> https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/test/utils/cryptography/ECDSA.test.js#L108
With `signature.v().to_u64() as u8` it can be also `0` or `1`.

Reply:
Oh, I see, I didn't expect that. Good one 👍